### PR TITLE
SC-1117 - Read api

### PR DESCRIPTION
**Pull request by Hari-stackroute**
Implemented read-api in RegistryHelper
Done pom changes in registry, renamed the registry jar to user-registry

### CODE REVIEW

File: java/registry/src/main/java/io/opensaber/registry/controller/RegistryController.java [hunk 1]
```diff
@@ -217,36 +216,17 @@
...
+            JsonNode resultNode = registryHelper.readEntity(inputJson,requireLDResponse);
             // Transformation based on the mediaType
             Data<Object> data = new Data<>(resultNode);
             Configuration config = configurationHelper.getResponseConfiguration(requireLDResponse);
 
             ITransformer<Object> responseTransformer = transformer.getInstance(config);
             Data<Object> resultContent = responseTransformer.transform(data);
             response.setResult(resultContent.getData());
-            logger.info("ReadEntity,{},{}", recordId.toString(), config);
+            logger.info("ReadEntity,{}", config);
```
Comment (indrajra) [inline java/registry/src/main/java/io/opensaber/registry/controller/RegistryController.java]: why is this removed? Remember, RegistryController is still going to be used by OS as a service...So we need it.

File: java/registry/src/main/java/io/opensaber/registry/helper/RegistryHelper.java [hunk 2]
```diff
@@ -63,5 +76,41 @@ public String addEntity(JsonNode inputJson, String userId) throws Exception {
...
     }
+
+    /**
+     * Get entity details from the DB and modifies data according to view template
+     * @param inputJson
+     * @param requireLDResponse
+     * @return
+     * @throws Exception
+     */
+    public JsonNode readEntity(JsonNode inputJson, boolean requireLDResponse) throws Exception {
```
Comment (indrajra) [inline java/registry/src/main/java/io/opensaber/registry/helper/RegistryHelper.java]: Create another overloaded method that takes only one param...Let user-service use that.

File: java/registry/src/main/java/io/opensaber/registry/app/OpenSaberApplication.java [hunk 3]
```diff
@@ -28,7 +28,13 @@ public static void main(String[] args) {
 		context = application.run(args);
 	}
 
-	public static ApplicationContext getContext() {
+
+	/** This method return non-web application context
+	 * @return context
+	 */
+	public static ApplicationContext getAppContext() {
```
Comment (indrajra) [inline java/registry/src/main/java/io/opensaber/registry/app/OpenSaberApplication.java]: If this is what is getting used, why need to change main method? Can we not live with what was existing earlier. Am saying we instantiate the application separately....try this if it makes sense.
main() -> getWebAppContext() => generates a web application context.
getAppContext() => generates a none application context.
Calling a main method from another function seems weird.

File: java/registry/src/main/java/io/opensaber/registry/controller/RegistryController.java (hunk 1 above)
Comment (Hari-stackroute) [reply java/registry/src/main/java/io/opensaber/registry/controller/RegistryController.java]: Set back this one, logging record id.

File: java/registry/src/main/java/io/opensaber/registry/helper/RegistryHelper.java (hunk 2 above)
Comment (Hari-stackroute) [reply java/registry/src/main/java/io/opensaber/registry/helper/RegistryHelper.java]: Created new method which can only access by user-service

File: java/registry/src/main/java/io/opensaber/registry/app/OpenSaberApplication.java (hunk 3 above)
Comment (Hari-stackroute) [reply java/registry/src/main/java/io/opensaber/registry/app/OpenSaberApplication.java]: Agreed, changes done. Now web-app will call main method defaulty, getAppContext have the own code which created context, now we are not calling main method from this one.

File: java/registry/src/main/java/io/opensaber/registry/controller/RegistryController.java [hunk 4]
```diff
@@ -217,36 +216,17 @@
...
+            JsonNode resultNode = registryHelper.readEntity(inputJson,requireLDResponse);
             // Transformation based on the mediaType
             Data<Object> data = new Data<>(resultNode);
             Configuration config = configurationHelper.getResponseConfiguration(requireLDResponse);
 
             ITransformer<Object> responseTransformer = transformer.getInstance(config);
             Data<Object> resultContent = responseTransformer.transform(data);
             response.setResult(resultContent.getData());
-            logger.info("ReadEntity,{},{}", recordId.toString(), config);
+            logger.info("ReadEntity,{},{}", config,resultNode.get(apiMessage.getRequest().getEntityType()).get(uuidPropertyName));
```
Comment (indrajra) [inline java/registry/src/main/java/io/opensaber/registry/controller/RegistryController.java]: revert as was original - you have flipped now.

File: java/registry/src/main/java/io/opensaber/registry/app/OpenSaberApplication.java [hunk 5]
```diff
@@ -16,19 +16,19 @@
...
 
 	public static void main(String[] args) {
-		SpringApplication application = new SpringApplication(OpenSaberApplication.class);
-		if(args.length != 0 && args[0].equals(WebApplicationType.NONE.toString())){
-			application.setWebApplicationType(WebApplicationType.NONE);
-		} else {
-			application.setWebApplicationType(WebApplicationType.SERVLET);
-		}
-
+		application.setWebApplicationType(WebApplicationType.SERVLET);
```
Comment (indrajra) [inline java/registry/src/main/java/io/opensaber/registry/app/OpenSaberApplication.java]: why is this still existing? This isn't doing much value as this is the default.

File: java/registry/src/main/java/io/opensaber/registry/helper/RegistryHelper.java [hunk 6]
```diff
@@ -50,10 +63,10 @@ public String addEntity(JsonNode inputJson, String userId) throws Exception {
         String jsonString = objectMapper.writeValueAsString(inputJson);
         try {
             logger.info("Add api: entity type and shard propery: {}", shardManager.getShardProperty());
-
-            Shard shard = shardManager.getShard(inputJson.get(entityType).get(shardManager.getShardProperty()));
+            Map<String, Object> shardIdValue = objectMapper.convertValue(inputJson.get(entityType),Map.class);
```
Comment (indrajra) [inline java/registry/src/main/java/io/opensaber/registry/helper/RegistryHelper.java]: Why a map is needed? Can't it be simple like this
String shardId = inputJson.get(entityType).get(shardManager.getShardProperty());

File: java/registry/src/main/java/io/opensaber/registry/helper/RegistryHelper.java (hunk 2 above)
Comment (indrajra) [reply java/registry/src/main/java/io/opensaber/registry/helper/RegistryHelper.java]: Isn't this silly to repeat. Why is the new method created not calling the readEntity(inputJson, false)?

Comment (Hari-stackroute) [reply java/registry/src/main/java/io/opensaber/registry/helper/RegistryHelper.java]: My apologies for this weird changes, changed accordingly.

File: java/registry/src/main/java/io/opensaber/registry/app/OpenSaberApplication.java (hunk 5 above)
Comment (Hari-stackroute) [reply java/registry/src/main/java/io/opensaber/registry/app/OpenSaberApplication.java]: Value is nothing here, but i thought it will be good if it has WebApplicationType.SERVLET, since for normal application-context we added WebApplicationType.NONE.

Comment (Hari-stackroute) [reply java/registry/src/main/java/io/opensaber/registry/app/OpenSaberApplication.java]: Now it is removed.

File: java/registry/src/main/java/io/opensaber/registry/controller/RegistryController.java (hunk 4 above)
Comment (Hari-stackroute) [reply java/registry/src/main/java/io/opensaber/registry/controller/RegistryController.java]: Reverted.

File: java/registry/src/main/java/io/opensaber/registry/helper/RegistryHelper.java (hunk 6 above)
Comment (Hari-stackroute) [reply java/registry/src/main/java/io/opensaber/registry/helper/RegistryHelper.java]: Here it is not considered about shardid, tried to convert to map and get shard from the map(from map default type is integer) but when we try to get from JsonNode it will IntJson.
For this change we modified getShard() in ShardAdvisor.
